src/main/java/gyro/google/dns/PolicyResource.java

```diff
@@ -0,0 +1,276 @@
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.api.client.util.Data;
+import com.google.api.services.dns.Dns;
+import com.google.api.services.dns.model.PoliciesPatchResponse;
+import com.google.api.services.dns.model.Policy;
+import com.google.api.services.dns.model.PolicyAlternativeNameServerConfig;
+import com.google.api.services.dns.model.PolicyNetwork;
+import gyro.core.GyroException;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Required;
+import gyro.google.Copyable;
+import gyro.google.GoogleResource;
+
+/**
+ * Creates a DNS Policy.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     google::dns-policy dns-policy-example
+ *         name: "dns-policy-example"
+ *         description: "DNS Policy Example"
+ *         network
+ *             network: $(google::compute-network network-example)
+ *         end
+ *     end
+ */
+@Type("dns-policy")
+public class PolicyResource extends GoogleResource implements Copyable<Policy> {
+
+    private DnsPolicyAlternativeNameServerConfig alternativeNameServerConfig;
+
+    private String description;
+
+    private Boolean enableInboundForwarding;
+
+    private Boolean enableLogging;
+
+    private String id;
+
+    private String name;
+
+    private List<DnsPolicyNetwork> network;
+
+    /**
+     * Sets an alternative name server for the associated networks. When specified, all DNS queries are forwarded to a name server that you choose. Names such as .internal are not available when an alternative name server is specified. The value may be ``null``.
+     *
+     * @subresource gyro.google.dns.DnsPolicyAlternativeNameServerConfig
+     */
+    // XXX: https://github.com/perfectsense/gyro/issues/190
+    @Updatable
+    public DnsPolicyAlternativeNameServerConfig getAlternativeNameServerConfig() {
+        return alternativeNameServerConfig;
+    }
+
+    public void setAlternativeNameServerConfig(DnsPolicyAlternativeNameServerConfig alternativeNameServerConfig) {
+        this.alternativeNameServerConfig = alternativeNameServerConfig;
+    }
+
+    /**
+     * A mutable string of at most 1024 characters associated with this resource for the user's convenience. Has no effect on the policy's function. The value may be ``null``.
+     */
+    @Required
+    @Updatable
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    /**
+     * Allows networks bound to this policy to receive DNS queries sent by VMs or applications over VPN connections. When enabled, a virtual IP address will be allocated from each of the sub- networks that are bound to this policy. The value may be ``null``.
```

Comment:
This doc is to help a user configure the value for the field. As someone cannot set null in the config this portion is not necessary. Unless a field is required, it is optional and so if not configured by a user, will behave accordingly.
```suggestion
     * Allows networks bound to this policy to receive DNS queries sent by VMs or applications over VPN connections. When enabled, a virtual IP address will be allocated from each of the sub- networks that are bound to this policy.
```